agent/api/ecsclient/client.go

```diff
@@ -288,20 +288,20 @@ func (client *APIECSClient) getCustomAttributes() []*ecs.Attribute {
 
 func (client *APIECSClient) SubmitTaskStateChange(change api.TaskStateChange) error {
 	// Submit attachment state change
-	if change.Attachments != nil {
+	if change.Attachment != nil {
```

Comment:
Doesn't need to be this particular PR right now, but I'd prefer if the attachment logic were factored out into a separate function; it took me a bit to see that this block has an early return and doesn't invoke the other state-updating logic.

```go
if change.Attachment != nil {
	return client.submitAttachmentTaskStateChange(change)
}
```